src/components/code-editor/index.js

```diff
@@ -168,7 +169,10 @@ module.exports = {
     },
     computed: {
         isWritable: function() {
-	        return this.currentFile.isWritable();
+            if (this.isFileWritable == null) {
```

Comment:
Can you explain why this is necessary?

Reply:
I was seeing "This file has already been modified, use the returned instance"

Reply:
consequence of the call to clean()

Reply:
Surely then we should be updating currentFile?

Reply:
yes - that happens on line 145

Reply:
then why is this necessary?

Reply:
steps
1. computed field is called for the first time, isWriteable() method called on currentFile
2. file contents loaded into editor.
3. user makes change and select save
At this point computed field is called again. Previously this would lead to the error seen above even though currentFile is updated after the call to overwriteFileJS().

Reply:
Hmm it still feels like an antipattern. I suggest we try and rework it to not call it on the old object during an update.

Reply:
I would use the same pattern we do in filesystem, watchers for new currentFile values update a writable property.

Reply:
I have removed the watcher function. It isn't required.